components/feature/sendtab/src/main/java/mozilla/components/feature/sendtab/SendTabUseCases.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.sendtab
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.concept.sync.Device
+import mozilla.components.concept.sync.DeviceCapability
+import mozilla.components.concept.sync.DeviceConstellation
+import mozilla.components.concept.sync.DeviceEventOutgoing.SendTab
+import mozilla.components.concept.sync.TabData
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.support.ktx.kotlin.crossProduct
+
+/**
+ * Contains use cases for sending tabs to devices related to the firefox-accounts.
+ */
+class SendTabUseCases(accountManager: FxaAccountManager) {
```

Comment:
These methods currently swallow any errors that may occur.

While there will be some non-actionable errors, I'd expect applications to at least handle network problems by display a "try again" toast, for example. In either case, I think we shouldn't pretend we succeeded when we either didn't do anything or actually failed. Especially if we failed due to a transient network error, that user may be able to correct themselves easily (e.g. by disconnecting from a bad wi-fi).

`sendEventToDeviceAsync` returns a Deferred<Boolean>, where `false` means "it didn't work". It's not a very rich error reporting API, but it's what you want to expose for a basic "try again later" UI experience.

Reply:
That's a fair point. Thanks for the catch! What I've done is to make the use cases into suspend functions since we have to combine the result of sending multiple tabs together before we can return it to the app in case of a failure.